src/autopick/autopick.cpp

```diff
@@ -126,15 +127,13 @@ void autopick_pickup_items(PlayerType *player_ptr, grid_type *g_ptr)
             continue;
         }
 
-        char out_val[MAX_NLEN + 20];
         GAME_TEXT o_name[MAX_NLEN];
         if (o_ptr->marked.has(OmType::NO_QUERY)) {
             continue;
         }
 
         describe_flavor(player_ptr, o_name, o_ptr, 0);
-        sprintf(out_val, _("%sを拾いますか? ", "Pick up %s? "), o_name);
-        if (!get_check(out_val)) {
+        if (!get_check(std::string(format(_("%sを拾いますか? ", "Pick up %s? "), o_name)).data())) {
```

Comment:
In calls to public hengband functions (like get_check()), I tried to wrap any arguments that are calls to format() with std::string() in case the function calls format() internally before it uses the argument (and to make it robust against future changes to the function that would do that).  For similar reasons, the latest forced push backed out some places where sprintf() was changed to format() because the function took a concptr argument which might be the direct result of a call to format().

Reply:
I agree that it is better to convert to std::string for safety.
However, it is also true that std::string().data() makes the source code more difficult to read.
As another issue, we may want to consider changing the return value of format from a pointer to an internally held static buffer to a std::string.
If the return value is passed directly to a function, it could be handled by making the argument side of the function std::string_view, in addition to adding .data().

Reply:
In the interest of safety, I should rework this pull request to:

1. Avoid the use of format() unless the changed function was already calling it in the current code.
2. If any of the remaining uses of format() are being passed as function arguments and there's already places in the current code where the same function is called with format() as an argument, then don't wrap format() with std::string() for readability.  Otherwise, either avoid using format() as an argument or accept the readability loss and wrap it with std::string.

Reply:
Because of the large size of this pull request and its tendency to conflict with others, it is better to merge it once. At this time, it appears that there is a conflict in one file, so please rebase and resolve the conflict.

After merging this pull request, I would like to make the following modifications:

1. change return value of format() function to std::string
2. change the signature of functions that directly give the return value of the format() function as an argument from concptr to std::string_view type
3. Unwrap std::string().data().